### PR TITLE
feat: support config patches at the serverclass level

### PR DESCRIPTION
**Pull request by rsmitty**
This PR allows configPatches to be specified for serverclasses

### CODE REVIEW

File: app/cluster-api-provider-sidero/controllers/metalmachine_controller.go [hunk 1]
```diff
@@ -240,6 +240,13 @@ func (r *MetalMachineReconciler) reconcileDelete(ctx context.Context, metalMachi
 		}
 
 		if err := r.Get(ctx, namespacedName, serverResource); err != nil {
+			// bail early if the server can't be fetch. this likely means we've deleted the server from underneath already.
+			if apierrors.IsNotFound(err) {
```
Comment (rsmitty) [inline app/cluster-api-provider-sidero/controllers/metalmachine_controller.go:244]: This just returns early to handle the case where we were trying to patch a server after it had already been cleaned up.

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**